### DISCUSSION TITLE
Allow some properties be excluded from record equality check

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/roslyn#54244] [Q&A]
I get that `records` are created to represent a single value, but they are also a very convenient and quick way to create a class.

It's say I have a `Country` class. It has `CountryCode` and `Name` properties. Each country is uniquely identified by the `CountryCode`, but if I create a `Country` record, `Name` will also be taken into account for equality check. Is there a way to avoid that?

**Comment by ryzngard** [comment]
You can use `Equals` if the record is sealed : https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUBQKIAeEYADgDYCmASiwMYD2yGABMmL8AvPxgsA7gAoARADl6LWXH6yAahCaplq2QHkEACxbJZASgDc1Oo1YcefQYVHipcxWF1rDJ5Ju1vA2NTX1MLa3JaemZ2Ll4BZABmVwkZWTYklTVs2QiKCnxiAE5pIVExZEIrQpKy4RFKpJrKcnwUgGcWLRZE+KdouzjHDGkigAZ+TxZVCf4AnVniSbDkcwoAbwp+Hf52/mBubiZ+KgBHVC0O6UHYhwSAfn5uELXt3a3yXe/+KAAzfjSF5+X4dcSoJhMdZfH47T6w2H4ADs/D+VxYkQR/AAvu9YXifsjnq8AHTTCpTJSY3a48jYoA===

https://stackoverflow.com/questions/64326511/custom-equality-check-for-c-sharp-9-records has some good information on this
